padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,216 @@
+    if training:
+        dataset = dataset.shuffle(reshuffle=True)
+
+    def Collate(batch):
+        batch = collate_fn(batch)
+        batch['image'] = np.array([tensor for tensor in batch['image']])
+        return batch
+
+    return dataset.prefetch(
+        num_workers=1, buffer_size=10*batch_size
+    ).batch(batch_size=batch_size).map(Collate)
+
+
+class VAEModel(pt.Model):
+    def __init__(self, activation_function, hidden_channels, kernel_size):
+        super().__init__()
+        self.net = VAE(activation_function, hidden_channels, kernel_size)
+
+    def forward(self, inputs):
+        images = inputs['image'][:, None, :, :]
+        out, mu, std = self.net(images)
+
+        return dict(
+            input=images,
+            mean=mu,
+            prediction=out,
+            std=std
+        )
+
+    def review(self, inputs, outputs):
+        images = rearrange(outputs['input'], 'b c x y -> b (c x y)')
+        prediction = rearrange(outputs['prediction'], 'b c x y -> b (c x y)')
+
+        mse = nn.functional.mse_loss(prediction, images)  # reconstruction error
+
+        q = Normal(loc=outputs['mean'], scale=outputs['std'])  # posterior distribution
+        mean = torch.zeros(outputs['mean'].shape[-1])
+        std = torch.ones(outputs['std'].shape[-1])
+        p = Normal(loc=mean, scale=std)  # prior distribution zero mean and unit variance
+        kld = gaussian_kl_divergence(q=q, p=p)  # regularisation error
+
+        loss = mse + torch.sum(kld)
+
+        return dict(
+            loss=loss,
+        )
+
+
+class VAE(nn.Module):
+    """ A simple Variational Autoencoder"""
+
+    def __init__(self, activation_function, hidden_channels, kernel_size):
+        super().__init__()
+        self.n_layer = 3
+        self.padding = 1
+        self.size_bottleneck = 28  # image size 28*28
+        stride = int(kernel_size/2)
+
+        for idx in range(self.n_layer):
+            self.size_bottleneck = int(
+                (self.size_bottleneck + 2 * self.padding - (kernel_size - 1) - 1) / stride + 1)
+
+        # encoder
+        self.conv_encoder = nn.Sequential(
+            torch.nn.Conv2d(in_channels=1, out_channels=hidden_channels, kernel_size=kernel_size, stride=stride,
+                            padding=self.padding),
+            activation_function(),
+            torch.nn.Conv2d(in_channels=hidden_channels, out_channels=hidden_channels, kernel_size=kernel_size,
+                            stride=stride, padding=self.padding),
+            activation_function(),
+            torch.nn.Conv2d(in_channels=hidden_channels, out_channels=1, kernel_size=kernel_size, stride=stride,
+                            padding=self.padding),
+            activation_function(),
+        )
+        self.linear_1 = torch.nn.Linear(in_features=self.size_bottleneck**2, out_features=1)
+        self.linear_2 = torch.nn.Linear(in_features=self.size_bottleneck**2, out_features=1)
```

Comment:
I think the naming is weird. I would have expected that size_bottleneck would be the size of the latent embedding. However, if I understand correctly, the latent dim is one here (which actually is a much smaller bottleneck).